### ISSUE TITLE
Supporting GRU any time soon?

### ISSUE THREAD

**Issue by rainjacket** [deeplearning4j/deeplearning4j#4145]
Is there any chance of supporting GRUs sometime soon?

I found this stack overflow thread https://stackoverflow.com/questions/45472557/gru-in-deeplearning4j that seems to assert that it's not on the roadmap due to being too similar to LSTM.

For my use case, GRU is preferable to LSTM. (Anyway other layers / loss functions / nonlinearities / regularization methods are also basically similar to each other.) Plus as a keras user, it would be nice to eventually support all or nearly all of keras layers for importing.

Is there still any philosophical objection here, or is it just a matter of somebody implementing the layer type?

**Comment by AlexDBlack** [comment]
It's more a case of having the time to write the implementation... it'll be quicker/easier once SameDiff (our ND4J autodiff library) is done.

Anyway, we certainly will add it eventually for Keras import.

Keras import aside, I'd prioritize implementing this before GRU: https://github.com/deeplearning4j/deeplearning4j/issues/4085

**Comment by raver119** [comment]
GRU is implemented in libnd4j.

**Comment by maxpumperla** [comment]
let's not close this until it's in dl4j _and_ keras import is enabled

**Comment by agibsonccc** [comment]
Closing — this issue is over 7 years old with no recent activity. The codebase has undergone a major rewrite. If this is still relevant, please open a new issue with current reproduction steps.
